### PR TITLE
Removed NLocal inital_state that is not a QuantumCircuit

### PR DESCRIPTION
**Pull request by 1ucian0**
PR #6585 deprecated the support for non-`QuantumCircuit` in the `initial_state` parameter of `NLocal`. That deprecation was introduced in 0.18 (released in July 2021) and it is time to remove it.

### CODE REVIEW

**Comment by Cryoris** [comment]
Oh yeah we should remove that! Could you update the typehint from `Any` to `QuantumCircuit` to reflect that only circuits are supported? 🙂

**Comment by 1ucian0** [comment]
> Oh yeah we should remove that! Could you update the typehint from `Any` to `QuantumCircuit` to reflect that only circuits are supported? 🙂

Good catch! Fixed in df18791c333dace1847b6c8c3024d86356acf789

File: qiskit/circuit/library/n_local/n_local.py [hunk 1]
```diff
@@ -660,7 +659,7 @@ def initial_state(self) -> Any:
         return self._initial_state
```
Comment (Cryoris) [inline qiskit/circuit/library/n_local/n_local.py:659]: I can't comment there but there's one `Any` left a few lines above from here 😄

Comment (1ucian0) [reply qiskit/circuit/library/n_local/n_local.py:659]: and I found a third one. Have a look to c49ff90dd87e7e27657fbc06af09bed91419fb94 to check it makes sense

**Pull request merged**